src/containers/sb-file-uploader.jsx

```diff
@@ -92,23 +91,29 @@ class SBFileUploader extends React.Component {
     handleChange (e) {
         const {
             intl,
-            isShowingWithoutId,
             loadingState,
-            projectChanged
+            projectChanged,
+            userOwnsProject
         } = this.props;
 
         const thisFileInput = e.target;
         if (thisFileInput.files) { // Don't attempt to load if no file was selected
             this.fileToUpload = thisFileInput.files[0];
 
-            // Allow upload to continue only after confirmation if the project
-            // has changed and is not showing with ID. If it has an ID, this operation
-            // does not currently overwrite that project, so it is safe to do without confirmation.
-            const uploadAllowed = (isShowingWithoutId && projectChanged) ?
-                confirm(intl.formatMessage(sharedMessages.replaceProjectWarning)) : // eslint-disable-line no-alert
-                true;
-
-            if (uploadAllowed) this.props.requestProjectUpload(loadingState);
+            // If user owns the project, or user has changed the project,
+            // we must confirm with the user that they really intend to replace it.
+            // (If they don't own the project and haven't changed it, no need to confirm.)
+            let uploadAllowed = true;
```

Comment:
I tried to go for legibility here. It's not the most concise code, but I hope this will minimize the chance that the logic will be misunderstood.